src/js/components/ConnectMetamask/MetamaskProvider.tsx

```diff
@@ -0,0 +1,241 @@
+
+        providerState
+          .request({ method: "eth_chainId" })
+          .then((result: string) => {
+            const chainId = parseInt(result);
+            acc = { ...acc, chainId };
+
+            setAccount(acc);
+          });
+      });
+  };
+
+  const changeChain = async (chainId: string) => {
+    if (!providerState)
+      throw new Error("providerState equals to " + providerState);
+
+    await providerState
+      .request({
+        method: "wallet_switchEthereumChain",
+        params: [{ chainId: chainId }],
+      })
+      .then((result) => {
+        console.log(result);
+      })
+      .catch(async (error) => {
+        if (error.code === -32002) {
+          alert("You have already made a request, please complete it");
+          return false;
+        } else if (error.code === 4902) {
+          const integerChainId = parseInt(chainId).toString();
+          const network =
+            NETWORKS_BY_CHAIN_ID[integerChainId as keyof INetworksByChainId];
+
+          await providerState
+            .request({
+              method: "wallet_addEthereumChain",
+              params: [
+                {
+                  chainId: chainId,
+                  chainName: network,
+                  rpcUrls: [
+                    RPC_URLS_BY_NETWORK[network as keyof IRpcUrlsByNetwork],
+                  ],
+                },
+              ],
+            })
+            .then(() => {
+              return true;
+            })
+            .catch((error) => {
+              console.log(error);
+              return false;
+            });
+        }
+
+        return false;
+      });
+
+    return true;
+  };
+
+  React.useEffect(() => {
+    if (!providerState) return;
+
+    const setDataAccount = (accountName: string) => {
+      saveProvider();
+      setMainContent("wallet");
+
+      updateAccountData(accountName);
+    };
+
+    const onAccountChanged = (accounts: string[]): void => {
+      if (accounts.length === 0) {
+        onDisconnect();
+        return;
+      }
+
+      setDataAccount(accounts[0]);
+    };
+
+    const onConnect = async (): Promise<void> => {
+      if (!providerState)
+        throw new Error("providerState equals to " + providerState);
+
+      const changedAccount: string[] = await providerState.request({
+        method: "eth_requestAccounts",
+      });
+      setDataAccount(changedAccount[0]);
+    };
+
+    const subscribeEvents = async (): Promise<void> => {
+      if (!providerState)
+        throw new Error("providerState equals to " + providerState);
+
+      await providerState.on("chainChanged", onConnect);
+      await providerState.on("accountsChanged", onAccountChanged);
+      await providerState.on("connect", onConnect);
+      await providerState.on("disconnect", onDisconnect);
+    };
+
+    const onProviderChanged = async () => {
+      if (!providerState)
+        throw new Error("providerState equals to " + providerState);
+
+      await subscribeEvents();
```

Comment:
add the catch block for handle an errors what you throw :)

Reply:
Done